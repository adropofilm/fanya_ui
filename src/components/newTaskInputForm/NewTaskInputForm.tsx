import { ReactElement, useState } from "react";
import styles from "./NewTaskInputForm.module.css";
import { addTask, getTasks } from "../../utility/api";
import { Status, Task } from "../../types/Task.types";

export type Props = {
  setTasks: (tasks: ReadonlyArray<Task>) => void;
};

export const NewTaskInputForm = ({ setTasks }: Props): ReactElement => {
  const [title, setTitle] = useState("");

  const task = {
    title: title,
    status: Status.OPEN,
  };

  const onFormSubmit = async (event: {
    preventDefault: () => void;
  }): Promise<void> => {
    event.preventDefault();
    try {
      const response = await addTask(task);
      setTitle("");

      if (response) {
        const tasks = await getTasks();
        setTasks(tasks);
      }
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <form
      id={styles["new-task-input-form"]}
      className="flex-column-center"
      onSubmit={onFormSubmit}>
      <input
        id={styles["new-task-input"]}
        placeholder="Type the name of your task..."
        onChange={(text): void => {
          setTitle(text.target.value);
        }}
        type="text"
        value={title}
        required
      />

      <button id={styles["new-task-btn"]} type="submit">
        create task
      </button>
    </form>
  );
};
