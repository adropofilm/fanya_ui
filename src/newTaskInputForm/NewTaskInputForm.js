import React, { useState, useEffect } from "react";
import axios from 'axios';
import './styles.css';
import { TextField } from "@mui/material";

const NewTaskInputForm = () => {
    const [taskString, setTask] = useState("")

    let body = {
        title: taskString
    };

    const onFormSubmit = (taskString) => {
        console.log(`TASK IS ${taskString}`)
        const search = async () => {
            const { data } = await axios.post("http://localhost:4000/tasks", {
                body
            })
        };
        search();
    }

    return (
        <form onSubmit={(onFormSubmit}>
            <input 
                onChange={(text) => {
                    setTask(text.target.value);
                }}
                type="text"
                value={taskString}
                required/>
        </form>
    )
}

export default NewTaskInputForm;