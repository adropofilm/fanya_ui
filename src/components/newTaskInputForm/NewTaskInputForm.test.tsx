import renderer from "react-test-renderer";
import { NewTaskInputForm } from "./NewTaskInputForm";

it("renders correctly", ()=> {
    const tree = renderer   
        .create(<NewTaskInputForm />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})