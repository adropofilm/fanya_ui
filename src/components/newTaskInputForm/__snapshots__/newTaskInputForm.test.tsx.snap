// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="flex-column-center"
  id="new-task-input-form"
  onSubmit={[Function]}
>
  <input
    id="new-task-input"
    onChange={[Function]}
    placeholder="Type the name of your task..."
    required={true}
    type="text"
    value=""
  />
  <button
    id="new-task-btn"
    type="submit"
  >
    create task
  </button>
</form>
`;
